web/src/games/mergers/game.ts

```diff
@@ -586,23 +579,43 @@ export const MergersGame: Game<IG> = {
 
       moves: { swapAndSellStock },
 
+      endIf: (G: IG) => {
+        if (G.merger === undefined || G.merger.swapAndSells === undefined) return false;
+        for (const player of Object.values(G.players)) {
+          if (G.merger.swapAndSells[player.id] === undefined) {
+            return false;
+          }
+        }
+        return true;
+      },
+
+      next: (G: IG) => {
+        if (!G.merger) {
```

Comment:
@vdfdev had it almost right: The `next` hook is only called after `onEnd` so that the `merger` attribute has been erased already when we come to this point. Originally, when this check was in the "next player" hook, this was before the `onEnd` call so that there was still one `mergingChain` left (even though that last chain was already dealt with).